packages/rulesets/src/spectral/az-common.ts

```diff
@@ -30,6 +42,138 @@ const ruleset: any = {
         function: falsy,
       },
     },
+    LroStatusCodesReturnTypeSchema: {
+      description: "The '200'/'201' responses of the long running operation must have a schema definition.",
+      message: "{{error}}",
+      severity: "error",
+      resolved: true,
+      formats: [oas2],
+      given: ["$[paths,'x-ms-paths'].*[put][?(@property === 'x-ms-long-running-operation' && @ === true)]^"],
+      then: {
+        function: lroStatusCodesReturnTypeSchema,
+      },
+    },
+    NamePropertyDefinitionInParameter: {
+      description: "A parameter must have a `name` property for the SDK to be properly generated.",
+      message: "{{error}}",
+      severity: "error",
+      resolved: true,
+      formats: [oas2],
+      given: ["$..[?(@property === 'parameters')]"],
```

Comment:
you can avoid using '..' to search, you can use [ '$.parameters.*' , "$.paths.*.parameters.*" "$.paths.*.*.paraemters.*"] to enumerate all the cases